### PR TITLE
CSPL-274: Update multisite test cases

### PR DESCRIPTION
**Pull request by pdhanoyasplunk**
### smoketest.go 

- Added new test case for Multisite Indexer Cluster 
- Modified existing Single site test case to use separate config for CM.

### deployment.go

- Added new methods for Multisite Indexer Cluster 
- Added new method for single site Indexer Cluster

### CODE REVIEW

File: test/smoke/smoke_test.go [hunk 1]
```diff
@@ -69,37 +69,58 @@ var _ = Describe("Smoke test", func() {
 	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
 		It("can deploy indexers and search head cluster", func() {
 
-			err := deployment.DeployCluster(deployment.GetName(), 3)
+			idxCount := 3
```
Comment (sgontla) [inline test/smoke/smoke_test.go:72]: Nit: indexerCout, just to avoid confusion with indexes?

File: test/smoke/smoke_test.go [hunk 2]
```diff
@@ -203,4 +224,81 @@ var _ = Describe("Smoke test", func() {
...
+				_ = deployment.GetInstance(deployment.GetName(), idc)
+				return idc.Status.ClusterMasterPhase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Ensure the indexers of all sites go to Ready phase
+			siteIndexerMap := map[string][]string{}
+			for site := 1; site <= siteCount; site++ {
+				siteName := fmt.Sprintf("site%d", site)
+				instanceName := fmt.Sprintf("%s-%s", deployment.GetName(), siteName)
+				siteIndexerMap[siteName] = []string{fmt.Sprintf("splunk-%s-indexer-0", instanceName)}
```
Comment (sgontla) [inline test/smoke/smoke_test.go:259]: In the future, if we are considering more than one indexer/site, I guess, this needs to be reworked.... Wondering if its worth to have at least the SF/RF number of indexers?

Comment (pdhanoyasplunk) [reply test/smoke/smoke_test.go:259]: @sgontla I will pull this code out to test util package in future and will refactor this code.

File: test/smoke/smoke_test.go (hunk 1 above)
Comment (pdhanoyasplunk) [reply test/smoke/smoke_test.go:72]: @sgontla I will resolve this next time I am touching this file.

**Pull request merged**